opencti-platform/opencti-graphql/src/database/ai-llm.ts

```
@@ -142,9 +171,23 @@ export const queryAi = async (busId: string | null, developerMessage: string | n
   }
 };
 
+// NLQ AI Query with LangChain's Chat Models
 export const queryNLQAi = async (promptValue: ChatPromptValueInterface) => {
-  if (!llm) {
-    throw UnsupportedError('Incorrect AI configuration', { enabled: AI_ENABLED, type: AI_TYPE, endpoint: AI_ENDPOINT, model: AI_MODEL });
+  if (!nlqModel) {
+    throw UnsupportedError('Incorrect AI configuration for NLQ', {
+      enabled: AI_ENABLED,
+      type: AI_TYPE,
+      endpoint: AI_ENDPOINT,
+      model: AI_MODEL,
+    });
+  }
+
+  try {
+    logApp.info('[NLQ] Querying AI model for structured output');
+    return await nlqModel.withStructuredOutput(OutputSchema).invoke(promptValue);
+  } catch (err) {
+    logApp.error('[NLQ] Error querying AI model', { cause: err });
+    const errorMessage = err instanceof Error ? err.message : String(err);
+    return `An error occurred while processing NLQ: ${errorMessage}`;
```

don't return anything in case of error, but rather throw an error

---

I tried something, but I'll let you validate, I'm not familiar with TypeScript yet :)